transaction/std/src/transaction_builder.rs

```diff
@@ -3104,4 +3105,505 @@ pub mod transaction_builder_tests {
+            let funding_input_amount = Amount::new(funding_input_amt, token_id);
+            let funding_output_amount = Amount::new(funding_output_amt, token_id);
+            let funding_change_output_amount =
+                Amount::new(funding_input_amt - funding_output_amt - fee, token_id);
+            let mut funding_transaction_builder = TransactionBuilder::new(
+                block_version,
+                Amount::new(fee, token_id),
+                fog_resolver.clone(),
+                funding_memo_builder,
+            )
+            .unwrap();
+
+            // Make sample input supply
+            let funding_input_credentials = get_input_credentials(
+                block_version,
+                funding_input_amount,
+                &sender,
+                &fog_resolver,
+                &mut rng,
+            );
+            funding_transaction_builder.add_input(funding_input_credentials);
+
+            // Fund gift code TxOut
+            funding_transaction_builder
+                .add_output(
+                    funding_output_amount,
+                    &sender_reserved_destinations.gift_code_subaddress,
+                    &mut rng,
+                )
+                .unwrap();
+
+            funding_transaction_builder
+                .add_change_output(
+                    funding_change_output_amount,
+                    &sender_reserved_destinations,
+                    &mut rng,
+                )
+                .unwrap();
+
+            let funding_tx = funding_transaction_builder.build(&mut rng).unwrap();
+
+            // The transaction should have exactly 2 outputs
+            assert_eq!(funding_tx.prefix.outputs.len(), 2);
+
+            let funding_output = funding_tx
+                .prefix
+                .outputs
+                .iter()
+                .find(|tx_out| {
+                    subaddress_matches_tx_out(&sender, GIFT_CODE_SUBADDRESS_INDEX, tx_out).unwrap()
+                })
+                .expect("Didn't find gift code funding output");
+
+            let funding_change_output = funding_tx
+                .prefix
+                .outputs
+                .iter()
+                .find(|tx_out| {
+                    subaddress_matches_tx_out(&sender, CHANGE_SUBADDRESS_INDEX, tx_out).unwrap()
+                })
+                .expect("Didn't gift code funding change output");
+
+            validate_tx_out(block_version, funding_output).unwrap();
+            validate_tx_out(block_version, funding_change_output).unwrap();
+
+            assert!(
+                subaddress_matches_tx_out(&sender, GIFT_CODE_SUBADDRESS_INDEX, funding_output)
+                    .unwrap()
+            );
+            assert!(subaddress_matches_tx_out(
+                &sender,
+                CHANGE_SUBADDRESS_INDEX,
+                funding_change_output
+            )
+            .unwrap());
```

Comment:
These are unnecessary since the `.find()` calls above would not find anything and `.expect` will end the test

Reply:
Indeeed